libpromises/expand.c

```diff
@@ -345,12 +345,60 @@ Rval ExpandPrivateRval(const EvalContext *ctx,
     return returnval;
 }
 
-static inline bool StartsWithVariableDataListReference(const char *str)
+/**
+ * Detects a variable expansion inside of a data/list reference, for example
+ * "@(${container_name})" or "@(prefix${container_name})" or
+ * "@(nspace:${container_name})".
+ *
+ * @note This function doesn't have to be bullet-proof, it only needs to
+ *       properly detect valid cases. The rest is left to the parser and code
+ *       expanding variables.
+ */
+static inline bool ContainsVariableDataListReference(const char *str)
```

Comment:
Maybe this function deserves a unit test?

Reply:
Yeah, but then it couldn't be `static`. :disappointed:

Reply:
@vpodzime We have other cases where we include the `.c` file in the unit test.